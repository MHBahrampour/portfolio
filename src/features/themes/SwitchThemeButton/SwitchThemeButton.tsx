import { useContext } from "react";

import DarkIcon from "@mui/icons-material/Brightness4";
import LightIcon from "@mui/icons-material/Brightness7";
import { IconButton, useTheme } from "@mui/material";

import { ColorContext } from "../ColorContext";

export const SwitchThemeButton = () => {
  const theme = useTheme();
  const colorMode = useContext(ColorContext);

  return (
    <IconButton onClick={colorMode.toggleColorMode} className="text-inherit">
      {theme.palette.mode === "dark" ? <LightIcon /> : <DarkIcon />}
    </IconButton>
  );
};
