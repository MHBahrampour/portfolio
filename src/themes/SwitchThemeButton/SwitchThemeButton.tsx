import { useContext } from "react";

import { motion } from "framer-motion";
import { RiMoonFoggyFill, RiSunFoggyFill } from "react-icons/ri";

import { IconButton, useTheme } from "@mui/material";

import { itemsDirectionMotion } from "@/utils/motions";

import { ColorContext } from "../ColorContext";

export function SwitchThemeButton() {
  const theme = useTheme();
  const colorMode = useContext(ColorContext);

  const handleClick = () => {
    colorMode.toggleColorMode();
    document.querySelector("html")?.classList.toggle("dark");
  };

  return (
    <IconButton
      onClick={handleClick}
      className="left-2 text-inherit"
      component={motion.button}
      {...itemsDirectionMotion(6, "RL")}
    >
      {theme.palette.mode === "dark" ? <RiSunFoggyFill /> : <RiMoonFoggyFill />}
    </IconButton>
  );
}
