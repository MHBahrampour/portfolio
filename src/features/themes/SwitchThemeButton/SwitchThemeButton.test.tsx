import renderer from "react-test-renderer";

import { SwitchThemeButton } from "./SwitchThemeButton";

describe("SwitchThemeButton", () => {
  it("checks SwitchThemeButton's snapshot", () => {
    const tree = renderer.create(<SwitchThemeButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
