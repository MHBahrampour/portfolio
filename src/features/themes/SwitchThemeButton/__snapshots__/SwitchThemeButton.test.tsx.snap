// Vitest Snapshot v1

exports[`SwitchThemeButton > checks SwitchThemeButton's snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium text-inherit css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="Brightness4Icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
    />
  </svg>
</button>
`;
